Storage/Service/AbstractUrlStorage.php

```diff
@@ -2,193 +2,89 @@
-        }
-
-        //in order to avoid a folder with a to big number of files in
-        if ($confirmed) {
-            $folderName .= $ds . substr($hash, 0, 3);
-        }
-
-        //create folder if missing
         if (!file_exists($folderName)) {
             mkdir($folderName, 0777, true);
         }
 
         return $folderName . $ds . $hash;
     }
 
-    /**
-     * @param string $hash
-     * @param string $cacheContext
-     * @return bool
-     */
-    public function head(string $hash, ?string $cacheContext = null): bool
+    public function head(string $hash): bool
     {
-        return file_exists($this->getPath($hash, $cacheContext));
+        return file_exists($this->getPath($hash));
     }
 
-    /**
-     * @param string $hash
-     * @param string $filename
-     * @param string $cacheContext
-     * @return bool
-     */
-    public function create(string $hash, string $filename, ?string $cacheContext = null): bool
+    public function create(string $hash, string $filename): bool
     {
-        return copy($filename, $this->getPath($hash, $cacheContext));
-    }
-
-    /**
-     * @return bool
-     */
-    public function supportCacheStore(): bool
-    {
-        return true;
+        return copy($filename, $this->getPath($hash));
     }
 
-    /**
-     * @return resource|bool|StreamInterface
-     */
-    public function read(string $hash, ?string $cacheContext = null, bool $confirmed = true)
+    public function read(string $hash, bool $confirmed = true): StreamInterface
     {
-        $out = $this->getPath($hash, $cacheContext, $confirmed);
+        $out = $this->getPath($hash, $confirmed);
         if (!file_exists($out)) {
-            return false;
+            throw new NotFoundHttpException($hash);
         }
 
-        return fopen($out, 'rb');
-    }
-
-    /**
-     * @deprecated
-     * @param string $hash
-     * @param null|string $context
-     * @return \DateTime|null
-     */
-    public function getLastUpdateDate(string $hash, ?string $context = null): ?\DateTime
-    {
-        @trigger_error("getLastUpdateDate is deprecated.", E_USER_DEPRECATED);
-        $path = $this->getPath($hash, $context);
-        if (file_exists($path)) {
-            $time = @filemtime($path);
-            if (!$time) {
-                return null;
-            }
-            $dateTime = \DateTime::createFromFormat('U', (string) $time);
-            if (!$dateTime) {
-                return null;
-            }
-            return $dateTime;
-        }
-        return null;
+        return new Stream(fopen($out, 'rb'));
     }
 
 
-    /**
-     * @return bool
-     */
     public function health(): bool
     {
         return is_dir($this->getBaseUrl());
     }
 
-    public function getSize(string $hash, ?string $cacheContext = null): ?int
+    public function getSize(string $hash, ?string $cacheContext = null): int
     {
         $path = $this->getPath($hash, $cacheContext);
```

Comment:
$cacheContext should no longer be passed to getPath?

Reply:
Done